### DISCUSSION TITLE
Using Lualatex

### DISCUSSION THREAD

**Discussion by quarkQuark** [leo-colisson/robust-externalize#4] [Q&A] [answered]
How would I use this package with Lualatex? It looks like this is supported, because the documentation mentions `/robExt/latex/use lualatex`. I presume I should add this somehow to the `tikz` preset. I have tried:

``` latex
\documentclass{article}
\usepackage{robust-externalize}
\begin{document}
\robExtConfigure{
    add to preset={tikz}{
        use lualatex,
    },
}
\cacheTikz
\begin{tikzpicture}
    \filldraw[black] (0,0) circle (2pt);
\end{tikzpicture}
\end{document}
```

This compiles properly with `latexmk -pdflatex -shell-escape`, but not with `latexmk -lualatex -shell-escape`. Swapping Lualatex for Xelatex also does not work, and nor does importing the `shellesc` package. I get the following error:

```
! Undefined control sequence.
<argument> ...CompilationCommand ^^J\pdfmdfivesum
                                                  {\l_robExt_result_str ^^J}...
l.13 \end
         {tikzpicture}
```

(I also tried replacing every occurrence of `pdflatex` in `robust-externalize.sty` with `lualatex`, which also makes no difference. It even still compiles with pdflatex, which surprises me.)

**Comment by tobiasBora** [comment]
Could you try to add
```

\def\pdfmdfivesum#1{%
 \directlua{tex.print(string.upper(md5.sumhexa("\luaescapestring{#1}")))}%
}
```

Sorry I don't have a computer to test.

**Reply by tobiasBora** [reply]
On XeLaTeX you want to use `\def\pdfmdfivesum{\mdfivesum}`, but I need to check if it is enough when accessing a computer.

**Reply by quarkQuark** [reply]
I'm away at the moment, but I'll have a look Monday or Tuesday.

**Comment by tobiasBora** [comment, marked as answer]
The bug is solved on master (using basically the fix proposed above), thanks for the report!

**Comment by quarkQuark** [comment]
Thank you! This now works perfectly (without `\usepackage{shellesc}`, and I think even without `use lualatex`)

**Reply by tobiasBora** [reply]
Great, thanks for the report! Yeah, `use lualatex` is used if you want to compile the pictures themself using lualatex (so if you do not enable `use lualatex`, but compile with `lualatex` it will compile the main document with `lualatex`, and the cached pictures with `pdflatex`).
